src/init.cpp

```diff
@@ -1770,6 +1770,13 @@ bool AppInitMain(NodeContext& node, interfaces::BlockAndHeaderTipInfo* tip_info)
         if (index == std::string::npos) {
             if (Lookup(bind_arg, bind_addr, GetListenPort(), false)) {
                 connOptions.vBinds.push_back(bind_addr);
+                if (IsBadPort(bind_addr.GetPort())) {
+                    InitWarning(strprintf(
+                        _("-bind request to listen on port %u. This port is considered \"bad\" and "
+                          "thus it is unlikely that any Bitcoin Core peers connect to it. See "
+                          "https://fetch.spec.whatwg.org/#port-blocking"),
```

Comment:
I'm a bit confused as to why bitcoin cares about what a brower's fetch api considers a "bad port", it's not clear what this even means? Maybe I'm missing something.

Reply:
looks like I was missing this context: https://github.com/bitcoin/bitcoin/pull/23306#issuecomment-947516736

Reply:
Just to get a starting list of bad ports I think. Can adjust from there if certain ports should be added or removed from that.

Reply:
Adding entries to the list may be undesirable, as it may mean that someone running a node on such a port silently stops getting connections.

Not a very serious concern, as this would happen spread over some amount of time anyway, but still a reason I think to err on the side of caution and start with a possibly overly expansive list here

Reply:
I'll still point out that the link provided doesn't define why a particular port is "bad". Following the link to get an explaination leads to: "A port is a bad port if it is listed in the first column of the following table". ok...??? 😕 I found this pretty confusing from a user perspective trying to understand this warning.

Reply:
For me, the following comment answers the question:

> "why am I seeing a failed SSH-login to my server on port 22 from your IP-address w.x.y.z?"

(from https://github.com/bitcoin/bitcoin/pull/23306#issuecomment-947516736)

There is some similarity between browser logic and Bitcoin node logic in this aspect where a malicious actor may trick a honest node to try to open a connection to a "victim" target:
* a honest user may be tricked to click specially crafted (malicious) links like e.g. http://area51.gov:22/
* a honest node may be tricked to try to open connection to `area51.gov:22` if that address gets gossiped over the network

Reply:
I understand that from reading the comment on github, but I was just pointing out from a user perspective there is no insight from the link in the warning in this commit. Perhaps it could be reworded: "See https://fetch.spec.whatwg.org/#port-blocking for a list of bad ports". Or perhaps a link to docs within bitcoin itself that lists our bad ports with an explanation of why they are bad.

Reply:
> There is some similarity between browser logic and Bitcoin node logic in this aspect where a malicious actor may trick a honest node to try to open a connection to a "victim" target:
> a honest web-browser user may be tricked to click specially crafted (malicious) links like e.g. http://area51.gov:22/
> a honest node may be tricked to try to open connection to area51.gov:22 if that address gets gossiped over the bitcoin network

If this is the reason for adding a list of bad ports then we can add lot of other port numbers. Right now there are only 80 ports in the list. Example: 3389

Also this goes against the main reason we are allowing p2p connections to non-default ports: https://github.com/bitcoin/bitcoin/pull/23542#discussion_r766685970

Reply:
> Or perhaps (eventually? sweat_smile ) a link to docs within bitcoin itself that lists our bad ports with an explanation of why they are bad.

Done.